src/libgit2/transports/httpclient.c

```diff
@@ -1151,6 +1159,43 @@ GIT_INLINE(int) client_read(git_http_client *client)
 static bool parser_settings_initialized;
 static http_settings_t parser_settings;
 
+static size_t git_http_parser_execute(http_parser_t *parser, const char* data, size_t len)
+{
+#ifdef USE_LLHTTP
+	llhttp_errno_t error;
+	size_t parsed_len;
+
+	/*
+	 * Unlike http_parser, which returns the number of parsed
+	 * bytes in the _execute() call, llhttp returns an error
+	 * code.
+	 */
+
+	if (data == NULL || len == 0) {
+		error = llhttp_finish(parser);
+	} else {
+		error = llhttp_execute(parser, data, len);
+	}
+
+	parsed_len = len;
```

Comment:
This isn't correct. The `len` field is the maximum data we can parse in one shot (the size of the `data` buffer), but it's not the amount we actually parsed.

Reply:
Right, but if it return HPE_OK, it will have read/parsed the full buffer. If there's some error, then it did not process it in full. This logic I got from node's usage of llhttp here: https://github.com/nodejs/node/blob/8a41d9b636be86350cd32847c3f89d327c4f6ff7/src/node_http_parser.cc#L826

Reply:
So, what happens when it reaches the end of the data to be parsed, but it's not a full buffer length of data?